pkg/lib/cockpit.js

```diff
@@ -3698,6 +3698,40 @@ function factory() {
         return self;
     };
 
+    cockpit.fslist = function (path, attrs) {
+        const self = {
+            members: { },
+        };
+
+        event_mixin(self, { });
+
+        self.on_message = function (_event, payload) {
+            const msg = JSON.parse(payload);
+            const { path, event } = msg;
+            delete msg.path;
+            delete msg.event;
+
+            if (event.deleted) {
+                delete self.members[path];
+            } else {
+                self.members[path] = msg;
+            }
+
+            self.dispatchEvent("changed");
```

Comment:
Does this mean there is no difference between the initial `changed` and subsequent updates?

Reply:
This seems very elegant I must say.